test/integration/VirtMemTests.cc

```diff
@@ -0,0 +1,325 @@
+#include "gtest/gtest.h"
+#include "simeng/OS/MemRegion.hh"
+#include "simeng/OS/Process.hh"
+#include "simeng/OS/SimOS.hh"
+#include "simeng/OS/Vma.hh"
+#include "simeng/version.hh"
+
+using namespace simeng::OS;
+
+namespace {
+const std::string configStr =
+    "{Core: {ISA: AArch64, Simulation-Mode: emulation, Clock-Frequency: 2.5, "
+    "Timer-Frequency: 100, Micro-Operations: True, "
+    "Vector-Length: 512, Streaming-Vector-Length: 512},"
+    "Process-Image: {Heap-Size: 100000, Stack-Size: 100000, Mmap-Size: "
+    "200000}, Simulation-Memory: {Size: 500000}, CPU-Info: "
+    "{Generate-Special-Dir: False}}";
+
+TEST(VirtMemTest, MmapSysCallNoAddressNoFile) {
+  Config::set(configStr.c_str());
```

Comment:
Config has a `set()` function that accepts `std::string` so no need for conversion to `c_str()`. Please check throughout file

Reply:
done

Reply:
passing in `Config::set(configStr)` rather than `configStr(configStr.c_str())` leads to the following error:
```
[SimEng:ModelConfig] Could not read {Core: {ISA: AArch64, Simulation-Mode: emulation, Clock-Frequency: 2.5, Timer-Frequency: 100, Micro-Operations: True, Vector-Length: 512, Streaming-Vector-Length: 512},Process-Image: {Heap-Size: 100000, Stack-Size: 100000, Mmap-Size: 200000}, Simulation-Memory: {Size: 500000}, CPU-Info: {Generate-Special-Dir: False}}

```

Reply:
Yes, I see my mistake now. The set() function that uses a string is for file paths, not the config files themsleves. Leave it as is